test/very_good_infinite_list_test.dart

```diff
@@ -1,517 +1,340 @@
+                children: [
+                  TextButton(
+                    key: key,
+                    onPressed: () => setState(() {}),
+                    child: const Text('REBUILD'),
+                  ),
+                  Expanded(
+                    child: InfiniteList(
+                      itemCount: itemCount,
+                      hasReachedMax: hasReachedMax,
+                      onFetchData: () => onFetchDataCalls++,
+                      itemBuilder: (_, i) => Text('$i'),
+                    ),
+                  ),
+                ],
+              );
+            },
+          ),
+        );
+
+        itemCount = 5;
+        hasReachedMax = false;
+        await rebuild();
+
+        expect(onFetchDataCalls, equals(1));
+      },
+    );
 
-      await tester.pumpApp(InfiniteList(
-        itemLoader: itemLoader,
-        builder: InfiniteListBuilder(success: (_, __) => const SizedBox()),
-      ));
+    testWidgets(
+      'renders ListView',
+      (tester) async {
+        await tester.pumpApp(
+          InfiniteList(
+            itemCount: 3,
+            onFetchData: emptyCallback,
+            itemBuilder: (_, i) => Text('$i'),
+          ),
+        );
+
+        expect(find.byType(ListView), findsOneWidget);
+      },
+    );
 
-      expect(itemLoaderCallCount, equals(1));
-    });
+    testWidgets(
+      'passes padding to internal ListView',
+      (tester) async {
+        const padding = EdgeInsets.all(16);
+
+        await tester.pumpApp(
+          InfiniteList(
+            padding: padding,
+            itemCount: 3,
+            onFetchData: emptyCallback,
+            itemBuilder: (_, i) => Text('$i'),
+          ),
+        );
+
+        final listView = tester.widget<ListView>(find.byType(ListView));
+        expect(listView.padding, equals(padding));
+      },
+    );
 
-    testWidgets('renders default loading widget by default', (tester) async {
-      await tester.pumpApp(InfiniteList(
-        itemLoader: (int limit, {int? start}) async => [],
-        builder: InfiniteListBuilder(success: (_, __) => const SizedBox()),
-      ));
+    testWidgets(
+      'renders items using itemBuilder',
+      (tester) async {
+        const itemCount = 50;
+        var itemBuilderCalls = 0;
+
+        await tester.pumpApp(
+          InfiniteList(
+            itemCount: itemCount,
+            hasReachedMax: true,
+            onFetchData: emptyCallback,
+            itemBuilder: (_, i) {
+              itemBuilderCalls++;
+              return Text('$i');
+            },
+          ),
+        );
+
+        expect(itemBuilderCalls, equals(itemCount));
+      },
+    );
 
-      expect(find.byKey(const Key('__default_loading__')), findsOneWidget);
-    });
+    testWidgets(
+      'renders separators in between items using separatorBuilder',
+      (tester) async {
+        const itemCount = 20;
+        const separatorCount = itemCount - 1;
+        var separatorBuilderCalls = 0;
```

Comment:
It seems that no assertions is being done on this variable, maybe we could remove it?

```suggestion
```

Reply:
Nvm, misread the line.